frontend/pkg/frontend/helpers_test.go

```diff
@@ -194,9 +203,13 @@ func TestCheckForProvisioningStateConflict(t *testing.T) {
 
 				parentResourceID := resourceID.GetParent()
 				if parentResourceID.ResourceType.Namespace == resourceID.ResourceType.Namespace {
-					parentDoc := database.NewResourceDocument(parentResourceID)
-					parentDoc.ProvisioningState = parentState
-					_ = frontend.dbClient.CreateResourceDoc(ctx, parentDoc)
+					mockDBClient.EXPECT().
+						GetResourceDoc(gomock.Any(), gomock.Any()).
+						DoAndReturn(func(ctx context.Context, resourceID *arm.ResourceID) (*database.ResourceDocument, error) {
```

Comment:
same - it's a static value; use `Return`. There's more examples like these in other tests as well below but I won't comment on them to keep this review from getting redundant ;)